staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -101,6 +102,30 @@ func UpdateResource(r rest.Updater, scope RequestScope, admit admission.Interfac
 				return newObj, mutatingAdmission.Admit(admission.NewAttributesRecord(newObj, oldObj, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
 			})
 		}
+		transformers = append(transformers, func(ctx context.Context, newObj, oldObj runtime.Object) (runtime.Object, error) {
+			if hasUID, err := hasUID(oldObj); err != nil {
+				return nil, err
+			} else if !hasUID && scope.Authorizer != nil {
+				createAttributes := authorizer.AttributesRecord{
+					User:            userInfo,
+					ResourceRequest: true,
+					Path:            req.URL.Path,
+					Verb:            "create",
+					APIGroup:        scope.Resource.Group,
+					APIVersion:      scope.Resource.Version,
+					Resource:        scope.Resource.Resource,
+					Subresource:     scope.Subresource,
```

Comment:
need to think through whether we need to handle subresources differently here... we don't create-on-update on subresources today, do we? should an update of a missing subresource escalate to a create and return a forbidden error, or return a not found error?

Reply:
I think we should start off taking the position that no subresource allows create-on-update. We can always relax later when there is a concrete need (I don't know of one).

Reply:
> I think we should start off taking the position that no subresource allows create-on-update. We can always relax later when there is a concrete need (I don't know of one).

I'd tend to agree.